examples/functions.pl

```diff
@@ -188,6 +191,41 @@ sub ssl_io {
         #   this return code, a resumed handshake has just completed."
         elsif ($rc == MATRIXSSL_HANDSHAKE_COMPLETE) {
             $handshakeIsComplete = 1;
+
+            # Sometimes clients use the "false start" technique - they send
+            # the first application data packet immediately after sending the
+            # handshake "Finished" message without waiting for the server's own
+            # handshake "Finished" message. This saves a rountrip and speeds the
+            # data transfer in some cases by 33%
+
+            my $rc2 = $ssl->false_start_received_data($buf);
+
+            # all the codes are already explained in this function
+            while (1) {
+                if ($rc2 == MATRIXSSL_APP_DATA) {
+                    $appIn .= $buf if (defined($buf));
```

Comment:
How `$buf` can be undefined if `$rc2 == MATRIXSSL_APP_DATA` - is `false_start_received_data()` actually works this way?